okhttp-client-handler/src/main/scala/com/softwaremill/sttp/okhttp/OkHttpClientHandler.scala

```diff
@@ -46,12 +46,13 @@ abstract class OkHttpClientHandler[R[_], S](client: OkHttpClient)
     builder.build()
   }
 
-  private def setBody(requestBody: RequestBody[S]): Option[OkHttpRequestBody] = {
-    requestBody match {
+  private def setBody[T](request: Request[T, S]): Option[OkHttpRequestBody] = {
+    request.body match {
       case NoBody => None
-      case StringBody(b, encoding, _) =>
-        Some(OkHttpRequestBody.create(MediaType.parse(encoding), b))
-      case ByteArrayBody(b, _) => Some(OkHttpRequestBody.create(null, b))
+      case StringBody(b, _, _) =>
+        Some(OkHttpRequestBody.create(null, b))
```

Comment:
One last thing - any reason why you no longer use the `encoding` here?

Reply:
Yes, about that , the thing is that `MediaType.parse(encoding)` always fail because of the missing information (like type and subtype). With the encoding only we can't create the `MediaType`

Reply:
Ah, that's a good reason :)